src/scripts/main.js

```diff
@@ -1,5 +1,28 @@
 'use strict';
+const wall = document.querySelector('.wall');
 
-document.addEventListener('click', (e) => {
-  // write code here
+wall.addEventListener('click', (e) => {
+  const spider = document.querySelector('.spider');
+  const spiderRect = spider.getBoundingClientRect();
+  const wallRect = wall.getBoundingClientRect();
+   const clickX = e.clientX - wallRect.left;
+   const clickY = e.clientY - wallRect.top;
+ 
+   let topPosition = clickY - (spiderRect.height / 2);
+   let leftPosition = clickX - (spiderRect.width / 2);
+
+  if (topPosition < 0) {
+    topPosition = 0;
+  } else if (topPosition > wallRect.height) {
+    topPosition = wallRect.height;
+  }
```

Comment:
The spider can move outside of the wall's boundaries at the bottom. The problem is that you are not subtracting the spider's height from the wall's height. This means that the spider's top position can be exactly at the wall's height, which would place the bottom of the spider outside of the wall. To fix this, you should subtract the spider's height from the wall's height when setting the maximum top position.